src/model_signing/signature/sigstore.py

```diff
@@ -0,0 +1,111 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Package provides the methods to sign and verify models with sigstore."""
+
+from typing import Optional
+
+from in_toto_attestation.v1 import statement
+from sigstore import dsse
+from sigstore import models as sig_models
+from sigstore import oidc
+from sigstore import sign
+from sigstore.verify import policy as sig_policy
+from sigstore.verify import verifier as sig_verifier
+from sigstore_protobuf_specs.dev.sigstore.bundle import v1 as bundle_pb
+
+from model_signing.signature.signing import Signer
+from model_signing.signature.verifying import VerificationError
+from model_signing.signature.verifying import Verifier
+
+
+class SigstoreSigner(Signer):
+    """Provides a Signer that uses sigstore for signing."""
+
+    CLIENT_ID = "sigstore"
+
+    def __init__(
+        self, disable_ambient: bool = True, id_provider: str | None = None
+    ):
+        token = self.__get_identity_token(disable_ambient, id_provider)
+        if not token:
+            raise ValueError("No identity token supplied or detected!")
+        self._id_provider = token.expected_certificate_subject
+        self._signing_id = token.identity
+
+        self._signer = sign.Signer(
+            identity_token=token, signing_ctx=sign.SigningContext.production()
+        )
```

Comment:
For testing, we should use staging, not production. This should be possible to do via constructor arguments